chain/network/src/raw/tests.rs

```diff
@@ -61,3 +63,63 @@ async fn test_raw_conn_pings() {
         }
     }
 }
+
+#[tokio::test]
+async fn test_raw_conn_state_parts() {
+    init_test_logger();
+    let mut rng = testonly::make_rng(33955575545);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let cfg = chain.make_config(rng);
+    let peer_id = cfg.node_id();
+    let addr = cfg.node_addr.unwrap();
+    let genesis_id = chain.genesis_id.clone();
+    let _pm = crate::peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        cfg,
+        chain,
+    )
+    .await;
+
+    let mut conn = raw::Connection::connect(
+        addr,
+        peer_id.clone(),
+        None,
+        &genesis_id.chain_id,
+        genesis_id.hash,
+        0,
+        time::Duration::SECOND,
+    )
+    .await
+    .unwrap();
+
+    let num_parts = 5;
+    let ttl = 100;
+    let block_hash = CryptoHash::from_str("5nPwPzNqLkbftPQ557Ewe84xjXDFaoFuKkKc62PsasKp").unwrap();
```

Comment:
maybe using `StateSync::get_epoch_start_sync_hash()` (or copy pasting it here if we dont wanna depend on `near_client`) would make this test easier to change. If anything changes about `data::Chain::make()` then this test would break

Reply:
As that block_hash is simply ignored, let me also ignore it :)
This test doesn't depend on `data::Chain::make()` but depends on handling of state part requests in the fake.